CONTRIBUTING.md

```diff
@@ -54,7 +52,151 @@ We have provided at profile for java located in [Codestyle File ./docs/CodeStyle
 - [IntelliJ](https://imagej.net/Eclipse_code_style_profiles_and_IntelliJ)
 - [Visual Studio Code](https://stackoverflow.com/questions/46030629/need-to-import-eclipse-java-formatter-profile-in-visual-studio-code)
 
-#### License
+## Commit Style
+
+SystemDS project has linear history throughout. Rebase cleanly and verify that the commit-SHA's
+of the Apache Repo are not altered.
+A general guideline is never to change a commit inside the systemds official repo, which is
+the standard practice. If you have accidentally committed a functionality or tag, let others know.
+And create a new commit to revert the functionality of the previous commit but **not force push**.
+
+
+### Tags
+
+The tags can be used in combination to one another. These are the only tags available.
+
+* `[MINOR]`: Small changesets with additional functionality
+
+  > Examples:
+  >
+  > This commit makes small software updates with refactoring
+  > 
+  > [`030fdab3`](https://github.com/apache/systemds/commit/030fdab3ebe6dedc3b4bb860e0ec5acfd9c38e5d) - `[MINOR] Added package to R dependency; updated Docker test image`
+  >
+  > This commit cleans up the redundant code for simplification
+  > 
+  > [`f4fa5650`](https://github.com/apache/systemds/commit/f4fa565013de13270df05dd37610382ca80f7354) - `[MINOR][SYSTEMDS-43] Cleanup scale builtin function (readability)`
+  >
+
+* `[SYSTEMDS-#]`: A changeset which will a specific purpose such as a bug, Improvement, 
+   New feature, etc. The tag value is found from [SystemDS jira issue tracker](https://issues.apache.org/jira/projects/SYSTEMDS/issues).
+   
+* `[DOC]` also `[DOCS]`: Changes to the documentation
+  
+* `[HOTFIX]`: Introduce a fix in the _hot_ running system.
```

Comment:
i don't like the description here, it should say that it is commits that change behavior on already released versions, rather than using slang such as hot.